### DISCUSSION TITLE
Keep template submodule pointed at latest commit of repository

### DISCUSSION THREAD

**Discussion by corykinney** [community/community#32914] [Other Feature Feedback, Questions, & Ideas] [answered]
Is it possible to have a git submodule in a template repository that is pointed at the latest commit in a target repository, or would it be necessary to update the template repository to keep the submodule up-to-date with the changes on the target repository? Alternatively, would it be best just to require that `git submodule update --remote --merge` be called whenever the template is used to initialize a new repository?

**Comment by airtower-luna** [comment, marked as answer]
A submodule always points at a specific commit in a repository, so yes, you'd need to update the template repository.

Updating on initialization would be one option, another would be to [use Dependabot to keep the submodule updated](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates) (use the `gitsubmodule` "ecosystem").

Note that either way the generated repositories may end up with different versions of the submodule.

**Reply by corykinney** [reply]
Yes, it was expected that different repositories would end up with different versions depending on when they are generated. The main goal would be for a generated repository to always start with the latest version of the submodule at the time of generation; after that, the submodule would only be updated manually to ensure breaking changes don't get included.

Dependabot seems like it could be a good solution, just checking daily for a change in version number and then creating a pull request (maybe with an action to auto-merge). My understanding is that this is implemented through a `.github/dependabot.yml` file. Will repositories generated from the template automatically include this file (and any GitHub actions in the template repository) and attempt to continue updating the submodule automatically?

EDIT: It looks like the workflow would be copied as well as there doesn't yet seem to be a way to exclude files from the generated repository https://github.com/orgs/community/discussions/22183

**Reply by airtower-luna** [reply]
> The main goal would be for a generated repository to always start with the latest version of the submodule at the time of generation; after that, the submodule would only be updated manually to ensure breaking changes don't get included.

I don't know if it's feasible for your use case, but you could see if you can add a workflow that checks compatibility with a changed submodule version. Then people could still get updates, with a warning if they'd need to adapt.
